# Aim of the project

This program generate a graph to see the distance of a pairwise alignment. 
You can put several genes in a fasta file as the exemple [several_fasta.fasta](https://github.com/lucasdavid85/gene_project/blob/master/several_fasta.fasta)


# Installation

## Clone the git

On the terminal write the command
```bash
git clone https://github.com/lucasdavid85/gene_project.git
```
## Install the packages with pip
```bash
pip install PyQT5
```
Make sure that you installed gephi in the root to use it. 
You can download [gephi_here](https://gephi.org/users/download/)


# Usage
To run this program you had to use this following command :
```bash
python3 Gene_Graph_Lucas_DAVID.py
```

Then chose the fasta file and you can acc√®s either to the graph or choose to start gephi to manage the graph
* The script create a folder with the name of the file without extention and put the graph figure in the folder with the gefx file (used for gephi)

# Result example 
To see the result of a fasta example, you can go to the folder [/several_fasta/](https://github.com/lucasdavid85/gene_project/blob/master/several_fasta/)
This is an example of a graph : 

![Result graph](https://github.com/lucasdavid85/gene_project/blob/master/several_fasta/Fig_fastas.png)

# Licence
Lucas DAVID, M2Bioinformatics
